include/util/multi_level_partition.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef OSRM_UTIL_MULTI_LEVEL_PARTITION_HPP
+#define OSRM_UTIL_MULTI_LEVEL_PARTITION_HPP
+
+#include "util/typedefs.hpp"
+#include <cstdint>
+
+namespace osrm
+{
+namespace util
+{
+
+using LevelID = std::uint8_t;
+using CellID = std::uint32_t;
+
+// Mock interface, can be removed when we have an actual implementation
+class MultiLevelPartition
```

Comment:
What is the idea behind making this a abstract base class?

Can't this be a concept/trait like interface with concrete implementations?
We don't need runtime polymorphism here, do we?